src/OrchardCore.Modules/OrchardCore.Autoroute/Services/AutorouteEntries.cs

```diff
@@ -147,7 +155,10 @@ private async Task RefreshEntriesAsync(AutorouteStateDocument state)
             {
                 if (_stateIdentifier != state.Identifier)
                 {
-                    var indexes = await Session.QueryIndex<AutoroutePartIndex>(i => i.Id > _lastIndexId).ListAsync();
+                    var indexes = await Session
+                        .QueryIndex<AutoroutePartIndex>(i => i.Id > _lastIndexId)
+                        .OrderBy(i => i.Id)
```

Comment:
Interesting, I'm intrigued at the sql that it produces, as to why it would fail without an order by?

But I believe you :)

Reply:
There is no sql query that fails, but when there are many items the results may not be ordered by the `Id` column, and we rely on this order to manage the lastId cursor that we use to only query new rows.

The result is that when 1st publishing an item, in place of retrieving only one row and process it, you may retrieve many ones. Also, to remove all previous entries of contained items of containers, we were doing a lookup on the whole `_paths` dico values, this on each entry to add, here we now do only one `.ToLookup()`.

For info, in my case a 1st publishing was taking around 5 minutes, after this 1st publishing the lastId was in sync and then no problem on following ones.

Reply:
Ah I see, because you use LastOrDefault() to determine. Makes sense. 
Merge when you are ready.